x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/SSLService.java

```diff
@@ -483,33 +496,77 @@ X509ExtendedTrustManager wrapWithDiagnostics(X509ExtendedTrustManager trustManag
                     .limit(2) // we only need to distinguishing between 0/1/many
                     .map(Entry::getKey)
                     .collect(Collectors.toUnmodifiableList());
+                final String name;
                 switch (names.size()) {
                     case 0:
-                        return "(unknown)";
+                        name = "(unknown)";
+                        break;
                     case 1:
-                        return names.get(0);
+                        name = names.get(0);
+                        break;
                     default:
-                        return "(shared)";
+                        name = "(shared)";
+                        break;
                 }
+                return name + " (with trust configuration: " + configuration.getTrustConfig() + ")";
             };
             trustManager = new DiagnosticTrustManager(trustManager, contextName, diagnosticLogger::warn);
         }
         return trustManager;
     }
 
-    public static Map<String, SSLConfiguration> getSSLConfigurations(Settings settings) {
+    public static Map<String, SslConfiguration> getSSLConfigurations(Environment env) {
+        return getSSLConfigurations(env, env.settings());
+    }
+
+    private static Map<String, SslConfiguration> getSSLConfigurations(Environment env, Settings settings) {
         final Map<String, Settings> sslSettingsMap = getSSLSettingsMap(settings);
-        final Map<String, SSLConfiguration> sslConfigurationMap = new HashMap<>(sslSettingsMap.size());
+        final Map<String, SslConfiguration> sslConfigurationMap = new HashMap<>(sslSettingsMap.size());
         sslSettingsMap.forEach((key, sslSettings) -> {
             if (key.endsWith(".")) {
                 // Drop trailing '.' so that any exception messages are consistent
                 key = key.substring(0, key.length() - 1);
             }
-            sslConfigurationMap.put(key, new SSLConfiguration(sslSettings));
+            try {
+                sslConfigurationMap.put(key, SslSettingsLoader.load(sslSettings, null, env, getKeyStoreFilter(key)));
```

Comment:
This is a big bit of the change (even though it's a small change).

 In X-Pack, `SSLConfiguration` knew how to build itself from `Settings`. ssl-config doesn't do that. It has a separate loader concept. So this is where we've switch from the X-Pack loading logic to the ssl-config one.

`SslSettingsLoader` is an X-Pack specific implementation of the abstract `SslConfigurationLoader` from ssl-config to handle a few X-pack details.
One of which is this new `getKeyStoreFilter` (see below).